state/state.go

```diff
@@ -279,6 +280,43 @@ func (s *State) GetValidators() (last *types.ValidatorSet, current *types.Valida
 	return s.LastValidators, s.Validators
 }
 
+// VerifyEvidence verifies the evidence fully by checking it is internally
```

Comment:
See my other comment about State... I think we should move these kinds of functions out of State and keep State a simple snapshot.  LMK what you think.

Reply:
Ok - I think that's a nice idea. All of the `LoadXxx` just need the `db`, but nothing else from the state.

This VerifyEvidence does require a bit more. Do we just pass in a state as an arg here?

Reply:
Let's deal with it in a separate PR: https://github.com/tendermint/tendermint/issues/1010